src/scratchOrgInfoApi.ts

```diff
@@ -0,0 +1,415 @@
+  }
+
+  logger.debug(`_authorize - orgConfig.loginUrl: ${authInfo.getFields().loginUrl}`);
+  logger.debug(`_authorize - orgConfig.instanceUrl: ${authInfo.getFields().instanceUrl}`);
+
+  return authInfo;
+};
+
+const checkOrgDoesntExist = async (scratchOrgInfo: Record<string, unknown>): Promise<void> => {
+  const usernameKey = Object.keys(scratchOrgInfo).find((key: string) =>
+    key ? key.toUpperCase() === 'USERNAME' : false
+  );
+  if (!usernameKey) {
+    return;
+  }
+
+  const username = getString(scratchOrgInfo, usernameKey);
+
+  if (username && username.length > 0) {
+    try {
+      await AuthInfo.create({ username: username.toLowerCase() });
+    } catch (error) {
+      const sfdxError = SfdxError.wrap(error as Error);
+      // if an AuthInfo couldn't be created that means no AuthFile exists.
+      if (sfdxError.name === 'NamedOrgNotFound') {
+        return;
+      }
+      // Something unexpected
+      throw error;
+    }
+    // An org file already exists
+    throw SfdxError.create('scratchOrgInfoApi', 'signup', 'C-1007');
+  }
+};
+
+/**
+ * This extracts orgPrefs/settings from the user input and performs a basic scratchOrgInfo request.
+ *
+ * @param scratchOrgInfo - An object containing the fields of the ScratchOrgInfo.
+ * @returns {*|promise}
+ */
+export const requestScratchOrgCreation = async (
+  hubOrg: Org,
+  scratchOrgRequest: ScratchDefinition,
+  settings: SettingsGenerator
+): Promise<RecordResult> => {
+  // If these were present, they were already used to initialize the scratchOrgSettingsGenerator.
+  // They shouldn't be submitted as part of the scratchOrgInfo.
+  delete scratchOrgRequest.settings;
+  delete scratchOrgRequest.objectSettings;
+
+  // We do not allow you to specify the old and the new way of doing post create settings
+  if (scratchOrgRequest.orgPreferences && settings.hasSettings()) {
+    // This is not allowed
+    throw new SfdxError('signupDuplicateSettingsSpecified');
+  }
+
+  // See if we need to migrate and warn about using old style orgPreferences
+  if (scratchOrgRequest.orgPreferences) {
+    settings.migrate(scratchOrgRequest);
+  }
+
+  const scratchOrgInfo = mapKeys(scratchOrgRequest, upperFirst, true);
+
+  await checkOrgDoesntExist(scratchOrgInfo); // throw if it does exists.
+  try {
+    return await hubOrg.getConnection().sobject('ScratchOrgInfo').create(scratchOrgInfo);
+  } catch (error) {
+    const jsForceError = error as JsForceError;
+    if (jsForceError.errorCode === 'REQUIRED_FIELD_MISSING') {
+      throw new SfdxError(messages.getMessage('signupFieldsMissing', [jsForceError.fields.toString()]));
+    }
+    throw jsForceError;
+  }
+};
+
+/**
+ * This retrieves the ScratchOrgInfo, polling until the status is Active or Error
+ *
+ * @param hubOrg
+ * @param scratchOrgInfoId - the id of the scratchOrgInfo that we are retrieving
+ * @param timeout - A Duration object
+ * @returns {BBPromise}
+ */
+export const pollForScratchOrgInfo = async (
```

Comment:
@mshanemc Any plans on improving user feedback during org create? Right now there is absolutely no feedback during org create, which is frustrating to the user and to us when trying to diagnose org create issues. It sure would be nice if org create had better feedback.

Reply:
I was thinking of letting the user know how the org create is progressing by using scratch org info status as a source and possibly combine that set with the post signup steps, auth and lightning domain verification.